lib/note-detail.jsx

```diff
@@ -34,22 +30,34 @@ export default React.createClass( {
 		return note && note.id;
 	},
 
-	componentWillReceiveProps: function( { note } ) {
-		this.setState( {
-			content: get( note, 'data.content', '' )
-		} );
+	componentWillReceiveProps: function() {
+		const { note } = this.props;
+
+		if ( ! this.isValidNote( note ) || ! this.noteEditor ) return;
+
+		if ( saveTimerId ) clearTimeout( saveTimerId );
```

Comment:
`clearTimeout()` is always safe to call, even if the timer handle is `null`. I like to avoid the conditionals just to make the code smoother